newton/tests/test_obb.py

```diff
@@ -0,0 +1,261 @@
+        # Create vertices of a 2x3x4 box centered at origin
+        vertices = np.array(
+            [
+                [-1, -1.5, -2],
+                [1, -1.5, -2],
+                [1, 1.5, -2],
+                [-1, 1.5, -2],
+                [-1, -1.5, 2],
+                [1, -1.5, 2],
+                [1, 1.5, 2],
+                [-1, 1.5, 2],
+            ],
+            dtype=np.float32,
+        )
+
+        # Test PCA method
+        tf_pca, extents_pca = compute_pca_obb(vertices)
+        assert_np_equal(np.array(tf_pca[0:3]), np.zeros(3), tol=1e-6)
+        # Check extents (half-dimensions)
+        expected_extents = np.array([1.0, 1.5, 2.0])
+        # Sort extents for comparison as order might vary
+        assert_np_equal(np.sort(np.array(extents_pca)), np.sort(expected_extents), tol=1e-4)
+
+        # Test inertia method
+        tf_inertia, extents_inertia = compute_inertia_obb(vertices)
+        assert_np_equal(np.array(tf_inertia[0:3]), np.zeros(3), tol=1e-4)
+        assert_np_equal(np.sort(np.array(extents_inertia)), np.sort(expected_extents), tol=1e-4)
+
+    def test_rotated_box(self):
+        """Test OBB computation on a rotated box."""
+        # Create a box and rotate it 45 degrees around Z axis
+        angle = np.pi / 4
+        cos_a, sin_a = np.cos(angle), np.sin(angle)
+        rotation = np.array([[cos_a, -sin_a, 0], [sin_a, cos_a, 0], [0, 0, 1]])
+
+        # Original box vertices (2x3x4)
+        box_vertices = np.array(
+            [
+                [-1, -1.5, -2],
+                [1, -1.5, -2],
+                [1, 1.5, -2],
+                [-1, 1.5, -2],
+                [-1, -1.5, 2],
+                [1, -1.5, 2],
+                [1, 1.5, 2],
+                [-1, 1.5, 2],
+            ],
+            dtype=np.float32,
+        )
+
+        # Rotate vertices
+        vertices = (rotation @ box_vertices.T).T
+
+        # Test both methods
+        tf_pca, extents_pca = compute_pca_obb(vertices)
+        tf_inertia, extents_inertia = compute_inertia_obb(vertices)
+
+        # Convert to numpy
+        extents_pca_np = np.array(extents_pca)
+        extents_inertia_np = np.array(extents_inertia)
+
+        # Both methods should find reasonable OBBs
+        volume_pca = 8.0 * extents_pca_np[0] * extents_pca_np[1] * extents_pca_np[2]
+        volume_inertia = 8.0 * extents_inertia_np[0] * extents_inertia_np[1] * extents_inertia_np[2]
+
+        # Volumes should be positive
+        self.assertGreater(volume_pca, 0)
+        self.assertGreater(volume_inertia, 0)
+
+        # Inertia method might find a different orientation, but volume shouldn't be
+        # drastically larger than the original box volume (24.0)
+        self.assertLess(volume_inertia, 100.0)  # Reasonable upper bound
+
+        # Centers should be at origin
+        assert_np_equal(np.array(tf_pca[0:3]), np.zeros(3), tol=1e-4)
+        assert_np_equal(np.array(tf_inertia[0:3]), np.zeros(3), tol=1e-4)
```

Comment:
Perhaps it is possible to test the orientation of `tf_inertia` as well? It should hopefully find a rotation of `np.pi / 4` about the Z axis or its 180 degree counterpart?